### ISSUE TITLE
SC2089 should not be used when shell is something else than bash

### ISSUE THREAD

**Issue by nihilus** [koalaman/shellcheck#1014]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2089
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh

#!bin/dash
args='-lh "My File.txt"'
ls $args

```

#### Here's what shellcheck currently says:
SC2089

#### Here's what I wanted or expected to see:
Nothing

**Comment by koalaman** [comment]
It should warn since this won't work on any shell, but it shouldn't be recommending an array.

**Comment by oliv3r** [comment]
still recommends useisng an array on the wiki or use eval.

So I'm not sure I follow though; If I do
```sh
var="-v 'dir:dir'"
program "${var}"
```
what exactly will happen? The warning suggests the quotes will be gobbled up somewhere along the way.
I would expect though, that in the end, the shell uses `program -v 'dir:dir'` which is exactly what I intended.

**Comment by nihilus** [comment]
@oliv3r that is never going to work, '-v' is being treated as part of a string and not a command line switch.

**Comment by koalaman** [comment]
Good point! The warning is still highly relevant, but the message will now say "Rewrite using set/"$@" or functions." for sh/dash.
